subtests/docker_cli/attach/attach.py

```diff
@@ -36,10 +36,34 @@ def wait_interactive_cmd(self):
                      % wait)
         time.sleep(wait)
 
+    def image_exist(self, image_name):
+        di = DockerImages(self)
+        image_list = DockerImages.get_dockerimages_list(di)
+        exist_status = DockerImages.filter_list_full_name(image_list,
+                                                          image_name)
+        return exist_status
+
+    def pull_image(self, image_name):
+        print "pull_image"
+        dkrcmd = AsyncDockerCmd(self, 'pull', [image_name],
+                                self.config['docker_timeout'],
+                                verbose=True)
+        self.loginfo("Executing background command: %s" % dkrcmd)
+        dkrcmd.execute()
+        while not dkrcmd.done:
+            self.loginfo("Pulling...")
+            time.sleep(3)
+        result = dkrcmd.wait()
+        self.failif(result.exit_status != 0,
+                    "Fail to download image %s"
+                    % image_name)
+
     def initialize(self):
         super(attach_base, self).initialize()
         self.sub_stuff['subargs'] = self.config['run_options_csv'].split(',')
         fin = DockerImage.full_name_from_defaults(self.config)
+        if self.image_exist(fin) == []:
+            self.pull_image(fin)
```

Comment:
Yeah, i agree with you after a more deeper play around, a [pretest](https://github.com/autotest/autotest-docker/blob/master/pretests/pretest_example/pretest_example.py) is enough for handle this kind of issue.  
But it may not fit for this case now, since it put the `AsyncDockerCmd(self, 'run', self.sub_stuff['subargs'], verbose=True)` in every base clase's `def initialize`. Yeah, run command executes in initialize.....
Thus i'd like to fix the case now to let it run successfully and use the [pretest](https://github.com/autotest/autotest-docker/blob/master/pretests/pretest_example/pretest_example.py) as the next stage plan. What about it ? :)